assets/apps/dashboard/src/Components/Plugin/InstallActivate.js

```diff
@@ -0,0 +1,182 @@
+/* global neveDash */
+import { get } from '../../utils/rest';
+import { __ } from '@wordpress/i18n';
+import { useState } from '@wordpress/element';
+import { Button, Dashicon } from '@wordpress/components';
+import { useEffect } from 'react';
+
+const InstallActivate = ({
+	name = '',
+	labels = [],
+	slug,
+	successActivation,
+	autoInstall = false,
+	smallButton = false,
+	pluginState,
+	activateURL,
+	description,
+	// pluginBasename,
+	// successUpdate,
+}) => {
+	const { getPluginStateBaseURL, pluginsURL } = neveDash;
+
+	const [installing, setInstalling] = useState(false);
+	const [activating, setActivating] = useState(false);
+	// const [updating, setUpdating] = useState(false);
+	const [error, setError] = useState(false);
+	const [currentState, setCurrentState] = useState(pluginState);
+
+	const [buttonLabels, setButtonLabels] = useState({
+		firstLabel: labels.firstLabel ?? false,
+		installing: labels.installing ?? `${__('Installing', 'neve')}...`,
+		activating: labels.activating ?? `${__('Activating', 'neve')}...`,
+		installActivate:
+			labels.installActivate ?? `${__('Install and Activate', 'neve')}`,
+		activate: labels.activate ?? `${__('Activate', 'neve')}`,
+		installed: labels.installed ?? `${__('Installed', 'neve')}`,
+	});
+
+	const getLabel = (type) => {
+		if (buttonLabels.firstLabel) {
+			return buttonLabels.firstLabel;
+		}
+
+		return buttonLabels[type].replace('{plugin}', name);
+	};
+
+	const hideFirstLabel = () => {
+		buttonLabels.firstLabel = false;
+		setButtonLabels(buttonLabels);
+	};
+
+	const installPlugin = () => {
+		setInstalling(true);
+		hideFirstLabel();
+		wp.updates.ajax('install-plugin', {
+			slug,
+			success: () => {
+				activatePlugin();
+			},
+			error: (e) => {
+				if ('folder_exists' === e.errorCode) {
+					activatePlugin();
+				} else {
+					setError(
+						e.errorMessage
+							? e.errorMessage
+							: __(
+									'Something went wrong while installing the plugin.'
+							  )
```

Comment:
Can be shortened to something like this - also we should add the textdomain:
```suggestion
						e.errorMessage || __( 'Something went wrong while installing the plugin.', 'neve')
```